necsim/plugins/tskit/src/tree/mod.rs

```diff
@@ -0,0 +1,86 @@
+use std::{collections::HashMap, convert::TryFrom, fmt, fs::OpenOptions, io};
+
+use serde::Deserialize;
+use tskit::{IndividualId, NodeId, TableCollection};
+
+use necsim_core::{
+    event::{DispersalEvent, SpeciationEvent},
+    landscape::IndexedLocation,
+    lineage::GlobalLineageReference,
+};
+use necsim_core_bond::NonNegativeF64;
+
+mod metadata;
+mod reporter;
+mod table;
+
+// An arbitrary genome sequence interval
+const TSK_SEQUENCE_MIN: f64 = 0.0_f64;
+const TSK_SEQUENCE_MAX: f64 = 1.0_f64;
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Deserialize)]
+#[serde(try_from = "TskitTreeReporterArgs")]
+pub struct TskitTreeReporter {
+    last_parent_prior_time: Option<(GlobalLineageReference, NonNegativeF64)>,
+    last_speciation_event: Option<SpeciationEvent>,
+    last_dispersal_event: Option<DispersalEvent>,
+
+    // Original (present-time) locations of all lineages
+    origins: HashMap<GlobalLineageReference, IndexedLocation>,
```

Comment:
If you think performance is something to worry about here, there are several drop-in replacements for the std HashMap that are much faster.

Reply:
Could you maybe point me to one? What I found for a different problem was that just changing the hash function could give a boost ... Though I don't think using a specific hash map would be the bottleneck here (it's more that I have to use one at all).

Reply:
[fnv](https://crates.io/crates/fnv) is mentioned in the O'Reilly "Programming Rust" book, and I've used it w/no issue.  The main difference (for most uses) is that it isn't crypto-strength, which is why it is also faster.

Reply:
Thanks for the recommendation!

Reply:
Resolved in ef8a6bcd7b9475e534ba689cb0218d684aa8529d